core/templates/dev/head/pages/exploration_editor/editor_tab/state_editor_hints.html

```diff
@@ -53,8 +53,9 @@
       <div ng-if="editabilityService.isEditableOutsideTutorialMode()">
         <button type="button"
                 class="btn btn-default btn-lg oppia-add-hint-button protractor-test-oppia-add-hint-button"
-                ng-click="openAddHintModal()">
-          + Add Hint
+                ng-click="openAddHintModal()"
+                ng-if="hintCount < 5">
+          + Add Hint (maximum 5)
```

Comment:
I don't think you need the "(maximum 5)" there -- it's visual clutter, and the message is clear anyway when the button disappears. Let's just call it Add Hint, as before.

Reply:
Ok, removed it.